### PR TITLE
feat: Add new vertex for RIN

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/react/src/vertex3.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+            fill={shapeOptions.fill}
+            stroke={shapeOptions.stroke}
+            strokeWidth={shapeOptions.strokeWidth}
+
+            padding={padding}
+            cornerRadius={cornerRadius}
+        ></TextBox>
+    </>;
+};
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:45]: As discussed Annotation + Icon should probable be the same thing?

Then this becomes a variant of TextBox - IconTextBox or similar?

File: packages/react/src/vertex3.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        stroke: "#555555",
+        strokeWidth: 1
+    },
+    textOptions = {
+        fontFamily: "Verdana",
+        fontSize: 12,
+        fill: "#555555"
+    }
+}) => {
+    return <>
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:29]: No need for the `<>` outer tag

File: packages/react/src/vertex3.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            >
+
+            </Annotation>
+        );
+    });
+    return <g transform={`translate(${x}, ${y})`}>
+            {arr}
+        </g>
+        ;
+};
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:91]: No reason why this couldn't just create an Array of `TextBox`s?

File: apps/icon_fonts/css/fontawesome.css [hunk 4]
```diff
@@ -0,0 +1,4522 @@
+/*!
```
Comment (GordonSmith) [inline apps/icon_fonts/css/fontawesome.css:1]: This file should be removed

File: apps/icon_fonts/css/solid.css [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/*!
```
Comment (GordonSmith) [inline apps/icon_fonts/css/solid.css:1]: This file should be removed

File: packages/graph/src/graph2/graph.ts [hunk 6]
```diff
@@ -573,6 +573,10 @@ export class Graph2 extends SVGZoomWidget {
             .join(
                 enter => enter.append("g")
                     .attr("class", "graphEdge")
+                    .style("stroke", this.edgeColor())
+                    .style("stroke-width", (o) => {
+                        return this.edgeStrokeWidth() + "px";
+                    })
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts:579]: Color and width should be inside update method.

File: packages/react/src/text.tsx [hunk 7]
```diff
@@ -52,9 +52,9 @@ export const Text: React.FunctionComponent<Text> = ({
     onTotalWidthUpdate(ts.width);
     onTotalHeightUpdate(parts.length * (height + 2) - 2);
 
-    const yOffset = -(totalHeight / 2) + height / 2;
+    const yOffset = -(totalHeight / 2) + (height / 2);
     const TextLines = parts.map((p, i) => {
-        return <g transform={`translate(0 ${yOffset + i * (height + 2)})`}>
+        return <g transform={`translate(0, ${yOffset + i * (height + 2)})`}>
```
Comment (GordonSmith) [inline packages/react/src/text.tsx:57]: The non comma version of translate is preferable

File: packages/react/src/vertex3.tsx [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+    const textShapeHeight = textHeight + (textPadding * 2) + (textboxStrokeWidth * 2);
+    const annotationArr = [];
+    annotationMeta.forEach(anno => {
+        const annoText = anno.text;
+        const annoShapeWidth = textShapeHeight;
+        fullAnnotationWidth += annoShapeWidth + annotationGutter;
+        const annoOffsetX = fullAnnotationWidth - (annoShapeWidth / 2);
+        annotationArr.push(
+            <g class="vertex3-anno" transform={`translate(${annoOffsetX}, ${annoOffsetY})`}>
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:75]: The non comma version of translate is preferable

File: packages/react/src/vertex3.tsx [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+                    cornerRadius={cornerRadius}
+                    strokeWidth={0}
+                />
+            </g>
+        );
+    });
+    if (annotationMeta.length > 0) {
+        fullAnnotationWidth += annotationGutter * (annotationMeta.length - 1);
+    }
+    const textElement = <g transform={`translate(${textOffsetX}, ${annoOffsetY})`}>
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:91]: The non comma version of translate is preferable

File: packages/react/src/vertex3.tsx [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+    </g>;
+    const iconHeight = icon.height || 20;
+    const iconStrokeWidth = icon.strokeWidth || 0;
+    const iconOffsetX = 0;
+    const iconOffsetY = - (iconHeight / 2) - (textShapeHeight / 2) - (annotationGutter * 2) - (iconStrokeWidth);
+    const subTextOffsetX = 0;
+    const subTextOffsetY = textShapeHeight + (annotationGutter * 2);
+    return <g>
+            <g
+                transform={`translate(${iconOffsetX}, ${iconOffsetY})`}
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:112]: The non comma version of translate is preferable

File: packages/react/src/vertex3.tsx [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+            >
+                <Icon
+                    {...icon}
+                />
+            </g>
+            <g
+                transform={`translate(${-fullAnnotationWidth / 2}, ${annoOffsetY})`}
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:119]: The non comma version of translate is preferable

File: packages/react/src/vertex3.tsx [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+            >
+                {textElement}
+                {annotationArr}
+            </g>
+            <g
+                transform={`translate(${subTextOffsetX}, ${subTextOffsetY})`}
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:125]: The non comma version of translate is preferable

File: packages/react/src/text.tsx [hunk 13]
```diff
@@ -88,30 +90,36 @@ export const TextBox: React.FunctionComponent<TextBox> = ({
...
         />
-        <Text
-            text={text}
-            fontFamily={fontFamily}
-            height={height}
-            fill={textFill}
-            onSizeUpdate={onTextSizeUpdate}
-        />
-    </>;
+        <g transform={`translate(0, ${textOffsetY})`}>
```
Comment (GordonSmith) [inline packages/react/src/text.tsx:112]: The non comma version of translate is preferable

File: packages/graph/src/graph2/graph.ts [hunk 14]
```diff
@@ -992,6 +1005,10 @@ export interface Graph2 {
     forceDirectedLinkDistance(_: number): this;
     forceDirectedLinkStrength(): number;
     forceDirectedLinkStrength(_: number): this;
+    edgeColor(): number;
+    edgeColor(_: number): this;
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts:1009]: edgeColor should be a string.

File: packages/react/src/vertex3.tsx [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+    annotationGutter = 2,
+    annotationMeta = [],
+    cornerRadius = 3,
+    icon = {},
+    subText = {}
+}) => {
+    icon = {
+        height: 50,
+        imageChar: "?",
+        imageFontFamily: "'Font Awesome 5 Free'",
```
Comment (GordonSmith) [inline packages/react/src/vertex3.tsx:46]: Should default to "FA 4.7"

File: packages/graph/src/graph2/graph.ts (hunk 6 above)
Comment (GordonSmith) [reply packages/graph/src/graph2/graph.ts:579]: @jbrundage also no need for a function for "stroke-width"

File: packages/react/src/vertex3.tsx (hunk 15 above)
Comment (jbrundage) [reply packages/react/src/vertex3.tsx:46]: Why not use the latest? We're leaving so many icons behind with 4.7.

File: packages/graph/src/graph2/graph.ts [hunk 16]
```diff
@@ -573,6 +573,8 @@ export class Graph2 extends SVGZoomWidget {
             .join(
                 enter => enter.append("g")
                     .attr("class", "graphEdge")
+                    .style("stroke", this.edgeColor())
+                    .style("stroke-width", this.edgeStrokeWidth() + "px")
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts]: Move to outside the join

File: packages/react/src/vertex3.tsx (hunk 15 above)
Comment (GordonSmith) [reply packages/react/src/vertex3.tsx]: I am good with that - but there is more involved than referencing it here!!!

File: packages/react/src/text.tsx (hunk 7 above)
Comment (GordonSmith) [reply packages/react/src/text.tsx:57]: @jbrundage - bump

**Comment by GordonSmith** [comment]
@jbrundage - still has font files in and need to remove a few TRANSLATE commas.

**Pull request merged**